llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -911,7 +1020,8 @@ struct DSEState {
 
         auto *MD = dyn_cast_or_null<MemoryDef>(MA);
         if (MD && MemDefs.size() < MemorySSADefsPerBlockLimit &&
-            (getLocForWrite(&I) || isMemTerminatorInst(&I)))
+            (getLocForWrite(&I) || isMemTerminatorInst(&I) ||
+             HasInitializesAttr(&I)))
```

Comment:
should this check the EnableInitializesImprovement before considering HasInitializesAttr ?

Reply:
Yes, it's better to check the `EnableInitializesImprovement` at the beginning as well. Done!